src/Symfony/Component/Mailer/Transport.php

```diff
@@ -169,6 +169,12 @@ private static function createTransport(string $dsn, EventDispatcherInterface $d
                 }
 
                 throw new LogicException(sprintf('The "%s" scheme is not supported for mailer "%s".', $parsedDsn['scheme'], $parsedDsn['host']));
+            case 'custom':
+                if (!$customTransport) {
+                    throw new \LogicException('You must specify the transport class when using the "custom" provider.');
+                }
+
+                return new $customTransport($user, $pass, $query, $client, $dispatcher, $logger);
```

Comment:
`$customTransport` is an instance of `TransportInterface`, not a class name.
```suggestion
                return $customTransport;
```

Reply:
Hmm, yes you are right (I changed it just befor pushing), but what I wanted to do is to keep the data from the dsn (user, pass, and query), so I should keep the new and arguments, put do it an other way in the container extension

Reply:
I have edited, what do you think of this: https://github.com/symfony/symfony/pull/31931/commits/0c1a607908fcc8dc4e890491365e4a03f6a45530 ?